app/controllers/products_controller.rb

```diff
@@ -0,0 +1,63 @@
+class ProductsController < ApplicationController
+  # before_action :require_login, except: [:index, :show]
+
+  def index
+    @products = Product.active_product
+  end
+
+  def show
+    @product = Product.find(params[:id])
+  end
+
+  def new
+    @product = Product.new
+  end
+
+  def create
+    @product = Product.new(user_params[:product])
+
+    if @product.save
+      redirect_to product_path(@product.id)
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @product = Product.find(params[:id])
+  end
+
+  def update
+    @product = Product.find(params[:id])
```

Comment:
You use this line mutliple times in this controller so this would be a good opportunity for cleanup